ctapipe/image/cleaning.py

```diff
@@ -2,16 +2,65 @@
 Image Cleaning Algorithms (identification of noisy pixels)
 """
 
-__all__ = ['tailcuts_clean', 'dilate']
+__all__ = ['tailcuts_clean', 'dilate', 'fact_image_cleaning', 'ImageCleaning']
 
 import numpy as np
 from scipy.sparse.csgraph import connected_components
-
-
-def tailcuts_clean(geom, image, picture_thresh=7, boundary_thresh=5,
-                   keep_isolated_pixels=False,
-                   min_number_picture_neighbors=0):
-
+from ctapipe.core import Component
+from traitlets import Dict, CaselessStrEnum
+
+
+class ImageCleaning(Component):
+    '''
+    A component to make image cleaning configurable using traitlets.
+    '''
+
+    key_type = CaselessStrEnum(
+        default_value='camera_id',
+        values=['camera_id', 'telescope_id'],
+        help='If the keys in settings are telescope ids or camera ids'
+    ).tag(config=True)
+    settings = Dict(
+        help='A dict mapping either camera_id or telescope_id to cleaning settings.'
+        ' This are keyword arguments for the actual cleaning function plus the'
+        ' the key algorithm which can be either "tailcuts" or "fact"'
+    ).tag(config=True)
+
+    def __call__(self, geom, image, arrival_times=None, telescope_id=None):
+
+        if self.key_type == 'camera_id':
+            kwargs = self.settings.get(geom.cam_id, {})
+        elif self.key_type == 'telescope_id':
+            if telescope_id is None:
+                raise ValueError(
+                    'telescope_id is required if using key_type="telescope_id"'
+                )
+            kwargs = self.settings.get(telescope_id, {})
+
+        algorithm = kwargs.pop('algorithm', 'tailcuts')
```

Comment:
Why is this not an argument to the __init__? Its a bit hidden like this

Reply:
Because this way, it can be configured *per telescope*